extension/test/unit/lib/magento/customer/syncItems-test.js

```diff
@@ -0,0 +1,96 @@
+const assert = require('assert')
+const step = require('../../../../../lib/magento/customer/getSyncItems')
+
+describe('getSyncItems', () => {
+  let context
+  let input
+
+  beforeEach(() => {
+    context = {}
+    input = {}
+  })
+
+  it('Should return correct arrays for adding one product', async () => {
+    input = {
+      mageProductsIds: [{ id: 10 }],
+      localProductsIds: [{ id: 20 }]
+    }
+
+    const expectedAddResult = [{ id: 20 }]
+    const diffObject = await step(context, input)
+
+    assert.deepStrictEqual(diffObject.addProductsIds, expectedAddResult)
+  })
+
+  it('Should return a empty array - same products', async () => {
+    input = {
+      mageProductsIds: [{ id: 20 }],
+      localProductsIds: [{ id: 20 }]
+    }
+
+    const expectedAddResult = []
+    const diffObject = await step(context, input)
+
+    assert.deepStrictEqual(diffObject.addProductsIds, expectedAddResult)
+  })
+
+  it('Should return a empty array - no local products', async () => {
+    input = {
+      mageProductsIds: [{ id: 20 }, { id: 30 }],
+      localProductsIds: []
+    }
+
+    const expectedAddResult = []
+    const diffObject = await step(context, input)
+
+    assert.deepStrictEqual(diffObject.addProductsIds, expectedAddResult)
+  })
+
+  it('Should return correct arrays for empty mage products', async () => {
+    input = {
+      mageProductsIds: [],
+      localProductsIds: [{ id: 20 }, { id: 30 }]
+    }
+
+    const expectedAddResult = [{ id: 20 }, { id: 30 }]
+    const diffObject = await step(context, input)
+
+    assert.deepStrictEqual(diffObject.addProductsIds, expectedAddResult)
+  })
+
+  it('Should return correct arrays current products ids - merge', async () => {
+    input = {
+      mageProductsIds: [{ id: 40 }],
+      localProductsIds: [{ id: 20 }, { id: 30 }]
+    }
+
+    const expectedAddResult = [{ id: 40 }, { id: 20 }, { id: 30 }]
+    const diffObject = await step(context, input)
+
+    assert.deepStrictEqual(diffObject.productIds, expectedAddResult)
+  })
+
+  it('Should return correct arrays current products ids - no mage products', async () => {
+    input = {
+      mageProductsIds: [],
+      localProductsIds: [{ id: 20 }, { id: 30 }]
+    }
+
+    const expectedAddResult = [{ id: 20 }, { id: 30 }]
+    const diffObject = await step(context, input)
+
+    assert.deepStrictEqual(diffObject.productIds, expectedAddResult)
+  })
+
+  it('Should return correct arrays current products ids - no local products', async () => {
```

Comment:
`Should return correct arrays current products ids` in all tests is a bit confusing. Are you trying to say `Should not return guest product ids`?